### PR TITLE
Add  support for new  Xray graph scan API

### PR DESCRIPTION
**Pull request by talarian1**
- [x] All [tests](https://github.com/jfrog/jfrog-idea-plugin/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
-----

### CODE REVIEW

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java [hunk 1]
```diff
@@ -95,48 +114,68 @@ public void refreshScanManagers() throws IOException {
...
     }
 
-    private void createScanManagers(Map<Integer, ScanManager> scanManagers, Set<Path> paths) throws IOException {
+    private ScanLogic createScanLogic() {
+        Logger log =Logger.getInstance();
+        XrayClient client = createXrayClientBuilder(GlobalSettings.getInstance().getServerConfig(), log).build();
+        try {
+            XrayScanCache scanCache = new XrayScanCache(project.getName(), HOME_PATH.resolve("cache"), log);
+
+            if (GraphScanLogic.isXrayVersionSupported(client.system().version())) {
```
Comment (eyalbe4) [inline src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java:137]: isXrayVersionSupported --> isSupportedInXrayVersion

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java [hunk 2]
```diff
@@ -95,48 +114,68 @@ public void refreshScanManagers() throws IOException {
...
+                return new GraphScanLogic(scanCache,log);
+            } else {
+                if (BulkScanLogic.isXrayVersionSupported(client.system().version())) {
+                    return new BulkScanLogic(scanCache, log);
+                }
+            }
+        } catch (IOException e) {
```
Comment (eyalbe4) [inline src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java:144]: Better not to catch the exception, but let it be thrown. We should strive to catch them in upper most part of the stack trace as possible. The code is "healthier" this way.

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java [hunk 3]
```diff
@@ -95,48 +114,68 @@ public void refreshScanManagers() throws IOException {
...
+    private ScanLogic createScanLogic() {
+        Logger log =Logger.getInstance();
+        XrayClient client = createXrayClientBuilder(GlobalSettings.getInstance().getServerConfig(), log).build();
+        try {
+            XrayScanCache scanCache = new XrayScanCache(project.getName(), HOME_PATH.resolve("cache"), log);
+
+            if (GraphScanLogic.isXrayVersionSupported(client.system().version())) {
+                return new GraphScanLogic(scanCache,log);
+            } else {
+                if (BulkScanLogic.isXrayVersionSupported(client.system().version())) {
```
Comment (eyalbe4) [inline src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java:140]: If we got here, it means that we already sent a "get version" request to Xray, and now we're sending it again. Let's sent this request only once in the beginning.

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java [hunk 4]
```diff
@@ -6,28 +6,38 @@
...
 
 import static com.jfrog.ide.common.log.Utils.logError;
+import static com.jfrog.ide.common.utils.XrayConnectionUtils.createXrayClientBuilder;
 
 /**
  * Created by yahavi
  */
 public class ScanManagersFactory {
 
+    private static final Path HOME_PATH = Paths.get(System.getProperty("user.home"), ".jfrog-idea-plugin");
```
Comment (eyalbe4) [inline src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java:40]: The above variable is retrieved multiple times in the code, so the following is duplicated - 
```
Paths.get(System.getProperty("user.home"), ".jfrog-idea-plugin");
```
Let's create a utility method or methods (in Utils), to retrieve all the needed directories under the user home.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: src/main/java/com/jfrog/ide/idea/ui/configuration/JFrogGlobalConfiguration.java [hunk 5]
```diff
@@ -136,7 +137,7 @@ private String checkXrayConnection() {
             Version xrayVersion = xrayClient.system().version();
 
             // Check version
-            if (!isXrayVersionSupported(xrayVersion)) {
+            if (!isSupportedInXrayVersion(xrayVersion)) {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/ui/configuration/JFrogGlobalConfiguration.java:140]: Looks like a compilation issue here

File: src/test/java/com/jfrog/ide/idea/scan/PypiScanManagerTest.java [hunk 6]
```diff
@@ -40,6 +44,7 @@
     private static final String TRANSITIVE_DEPENDENCY_NAME = "anytree";
     private static final String TRANSITIVE_DEPENDENCY_VERSION = "2.8.0";
 
+
```
Comment (yahavi) [inline src/test/java/com/jfrog/ide/idea/scan/PypiScanManagerTest.java:47]:
```suggestion
```

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java [hunk 7]
```diff
@@ -45,7 +54,7 @@ private ScanManagersFactory(@NotNull Project project) {
     }
 
     /**
-     * Start an Xray scan for all projects.
+     * Start a Xray scan for all projects.
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java:57]:
```suggestion
     * Start an Xray scan for all projects.
```

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java [hunk 8]
```diff
@@ -95,48 +108,66 @@ public void refreshScanManagers() throws IOException {
...
+        XrayClient client = createXrayClientBuilder(GlobalSettings.getInstance().getServerConfig(), log).build();
+        Version xrayVersion = client.system().version();
+
+        if (GraphScanLogic.isSupportedInXrayVersion(xrayVersion)) {
+            return new GraphScanLogic(scanCache, log);
+        } else {
+            if (ComponentSummaryScanLogic.isSupportedInXrayVersion(xrayVersion)) {
+                return new ComponentSummaryScanLogic(scanCache, log);
+            }
+        }
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java:138]:
```suggestion
        } 
        if (ComponentSummaryScanLogic.isSupportedInXrayVersion(xrayVersion)) {
            return new ComponentSummaryScanLogic(scanCache, log);
        }
```

File: src/main/java/com/jfrog/ide/idea/configuration/ServerConfigImpl.java [hunk 9]
```diff
@@ -232,6 +242,11 @@ void setExcludedPaths(String excludedPaths) {
         this.excludedPaths = excludedPaths;
     }
 
+    void setProject(String project) {
+        this.project = project;
+    }
+
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/configuration/ServerConfigImpl.java:248]:
```suggestion
```

File: src/main/java/com/jfrog/ide/idea/utils/Utils.java [hunk 10]
```diff
@@ -17,6 +17,9 @@
  */
 public class Utils {
 
+    public static final Path HOME_PATH = Paths.get(System.getProperty("user.home"), ".jfrog-idea-plugin");
+
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/utils/Utils.java:21]:
```suggestion
```

**Review by yahavi — CHANGES_REQUESTED** [review]
* Please make sure to run reformat and organize imports on all changed files (I missed some in the review)
* The project configuration should be also reflected in ConfigurationTest

**Pull request merged**